bluebottle/activities/serializers.py

```diff
@@ -0,0 +1,67 @@
+from rest_framework import serializers
+from rest_framework_json_api.relations import ResourceRelatedField
+from rest_framework_json_api.serializers import ModelSerializer
+
+from bluebottle.activities.models import Activity
+from bluebottle.members.serializers import UserPreviewSerializer
+from bluebottle.utils.fields import FSMField
+from bluebottle.utils.serializers import (
+    ResourcePermissionField, FSMSerializer
+)
+
+
+class BaseActivitySerializer(ModelSerializer):
+    id = serializers.CharField(source='slug', read_only=True)
+    status = FSMField(read_only=True)
+    permissions = ResourcePermissionField('activity-detail', view_args=('slug',))
+    type = serializers.SerializerMethodField()
+    owner = ResourceRelatedField(read_only=True)
+
+    def get_type(self, instance):
+        return instance._meta.model_name
+
+    class Meta:
+        model = Activity
+        fields = (
+            'id',
+            'initiative',
+            'status',
+            'owner',
+            'title',
+            'description',
+            'type'
+        )
+
+    included_serializers = {
+        'initiative': 'bluebottle.initiatives.serializers.InitiativeSerializer',
+        'owner': 'bluebottle.initiatives.serializers.MemberSerializer',
+    }
+
+    class JSONAPIMeta:
+        included_resources = [
+            'owner',
+            'initiative'
+        ]
+        resource_name = 'activities'
+
+
+class ActivitySerializer(BaseActivitySerializer):
```

Comment:
I think we should use a polymorphic serializer here instead of this stuff. And I remember implementing this actually.